generators/Exercise.cs

```diff
@@ -1,23 +1,74 @@
-﻿using Generators.Input;
+﻿using System.Collections.Generic;
+using System.Linq;
+using Generators.Input;
 using Generators.Output;
 
 namespace Generators
 {
     public abstract class Exercise
     {
-        protected Exercise()
+        private static readonly ExerciseWriter ExerciseWriter = new ExerciseWriter();
+        
+        public string Name => GetType().ToExerciseName();
+
+        protected CanonicalData CanonicalData { get; private set; }
+        
+        public void Regenerate(CanonicalData canonicalData)
+        {
+            CanonicalData = canonicalData;
+            UpdateCanonicalData(CanonicalData);
+
+            ExerciseWriter.WriteToFile(this);
+        }
+
+        protected virtual void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+        }
+
+        public string Render() => RenderTestClass();
+
+        protected virtual string RenderTestClass() => CreateTestClass().Render();
+
```

Comment:
I might be missing the bigger picture here, of why these were split up, but it should be possible to just remove Render and update the call in ExerciseWriter so everyone is using RenderTestClass.